autoPyTorch/datasets/base_dataset.py

```diff
@@ -13,36 +13,36 @@
 
 from autoPyTorch.constants import CLASSIFICATION_OUTPUTS, STRING_TO_OUTPUT_TYPES
 from autoPyTorch.datasets.resampling_strategy import (
-    CROSS_VAL_FN,
+    CrossValFunc,
+    CrossValFuncs,
     CrossValTypes,
     DEFAULT_RESAMPLING_PARAMETERS,
-    HOLDOUT_FN,
-    HoldoutValTypes,
-    get_cross_validators,
-    get_holdout_validators,
-    is_stratified,
+    HoldOutFunc,
+    HoldOutFuncs,
+    HoldoutValTypes
 )
 from autoPyTorch.utils.common import FitRequirement, hash_array_or_matrix
 
-BASE_DATASET_INPUT = Union[Tuple[np.ndarray, np.ndarray], Dataset]
+BaseDatasetInputType = Union[Tuple[np.ndarray, np.ndarray], Dataset]
 
 
 def check_valid_data(data: Any) -> None:
-    if not (hasattr(data, '__getitem__') and hasattr(data, '__len__')):
+    if not all(hasattr(data, attr) for attr in ['__getitem__', '__len__']):
         raise ValueError(
-            'The specified Data for Dataset does either not have a __getitem__ or a __len__ attribute.')
+            'The specified Data for Dataset must have both __getitem__ and __len__ attribute.')
 
 
-def type_check(train_tensors: BASE_DATASET_INPUT, val_tensors: Optional[BASE_DATASET_INPUT] = None) -> None:
-    for i in range(len(train_tensors)):
-        check_valid_data(train_tensors[i])
+def type_check(train_tensors: BaseDatasetInputType, val_tensors: Optional[BaseDatasetInputType] = None) -> None:
+    for train_tensor in train_tensors:
+        check_valid_data(train_tensor)
     if val_tensors is not None:
         for i in range(len(val_tensors)):
             check_valid_data(val_tensors[i])
 
 
 class TransformSubset(Subset):
     """
+    I did not understand the meaning of this doc-string.
```

Comment:
Let us change the statement here. Maybe something like this (please change as you see fit)?

```
This class extends the PyTorchs' Subset class with the ability to select among train and test data. 
Fundamentally, the main dataset class (BaseDataset)  requires selecting which transformations are to be applied to the data, when accessing an item within the dataset.
 The TransformSubset allows an extra boolean while indexing the main dataset towards this goal.
```

Reply:
I changed the doc-string in the `refactor_development` [here](https://github.com/automl/Auto-PyTorch/blob/refactor_development/autoPyTorch/datasets/base_dataset.py#L45).
Could you compare and make it better?